tests/js/core/widget/core/Spin/Spin.js

```diff
@@ -1,181 +1,352 @@
+			spinWidget.value(5);
+			equal(spinWidget.value(), 5,
+				"Default option value of Spin is " + 5);
+			equal(typeof spinWidget.option("labels"), typeof defaultoptions.labels,
+				"Type of Default option labels of Spin is " + typeof defaultoptions.labels);
+
+			spinWidget.destroy();
+		});
+
+		test("Defined options Spin test", 14, function () {
+			var spin = document.getElementById("spin"),
+				options = {
+					min: 3,
+					max: 10,
+					moduloValue: "enabled",
+					shortPath: "enabled",
+					duration: 200,
+					direction: "down",
+					rollHeight: "container", // "container" | "item" | "custom"
+					itemHeight: 48,
+					momentumLevel: 3, // 0 - one item on swipe
+					scaleFactor: 0.7,
+					moveFactor: 0.1,
+					loop: "disabled",
+					labels: ["Aaa", "Bbb", "Ccc", "Ddd"],
+					digits: 3, // 0 - doesn't complete by zeros
+					dragTarget: "self"
+				},
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin", options);
+
+			equal(spinWidget.option("min"), options.min, "Option min of Spin is " + options.min);
+			equal(spinWidget.option("max"), options.max, "Option max of Spin is " + options.max);
+			equal(spinWidget.option("moduloValue"), options.moduloValue,
+				"Option moduloValue of Spin is " + options.moduloValue);
+			equal(spinWidget.option("shortPath"), options.shortPath,
+				"Option shortPath of Spin is " + options.shortPath);
+			equal(spinWidget.option("duration"), options.duration,
+				"Option duration of Spin is " + options.duration);
+			equal(spinWidget.option("rollHeight"), options.rollHeight,
+				"Option rollHeight of Spin is " + options.rollHeight);
+			equal(spinWidget.option("itemHeight"), options.itemHeight,
+				"Option itemHeight of Spin is " + options.itemHeight);
+			equal(spinWidget.option("momentumLevel"), options.momentumLevel,
+				"Option momentumLevel of Spin is " + options.momentumLevel);
+			equal(spinWidget.option("scaleFactor"), options.scaleFactor,
+				"Option scaleFactor of Spin is " + options.scaleFactor);
+			equal(spinWidget.option("moveFactor"), options.moveFactor,
+				"Option moveFactor of Spin is " + options.moveFactor);
+			equal(spinWidget.option("loop"), options.loop,
+				"Option loop of Spin is " + options.loop);
+			equal(spinWidget.option("labels"), options.labels,
+				"Option labels of Spin is " + options.labels);
+			equal(spinWidget.option("digits"), options.digits,
+				"Option digits of Spin is " + options.digits);
+			equal(spinWidget.option("dragTarget"), options.dragTarget,
+				"Option dragTarget of Spin is " + options.dragTarget);
+
+			spinWidget.destroy();
+		});
+
+		test("Undefined values of options Spin test", 4, function () {
+			var spin = document.getElementById("spin"),
+				options = {
+					min: undefined,
+					max: undefined,
+					duration: undefined
+				},
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin", options);
+
+			spinWidget.value(undefined);
```

Comment:
how about having check for undefined too?
equal(spinWidget.value(), undefined,"Undefined check");

Reply:
This `undefied` doesn't make sense. I'll remove this.